doc/contributing/adding-v8-fast-api.md

```diff
@@ -0,0 +1,150 @@
+# Adding V8 Fast API
+
+Node.js uses [V8](https://v8.dev/) as its JavaScript engine.
+Embedder functions implemented in C++ incur a high overhead, so V8
+provides an API to implement fast-path C functions which may be invoked directly
+from JITted code. These functions also come with additional constraints,
+for example, they may not trigger garbage collection.
+
+## Limitations
+
+* Fast API functions may not trigger garbage collection. This means by proxy
+  that JavaScript execution and heap allocation are also forbidden, including
+  `v8::Array::Get()` or `v8::Number::New()`.
+* Throwing errors is not available on fast API, but can be done
+  through the fallback to slow API.
```

Comment:
Perhaps it's also better to suggest that since in principal, the fast API and the slow API should behave the same to avoid bugs, it's better not to throw errors in the slow API either and just update some flags and return to the JS land to throw?

Reply:
I agree in principal with @joyeecheung, but that is not a limitation of a V8 Fast API declaration, but an implementation recommendation. I think it should not be included in this documentation, but in a more `style` specific documentation like README.md?